File: Internationalization/SetTextI18n/src/main/java/net/twisterrob/lints/internationalization/SetTextI18nViolation.java
package net.twisterrob.lints.internationalization;

import android.widget.TextView;

public class SetTextI18nViolation {

	void literal(TextView textView) {
		textView.setText("Hello");
	}

	void numberFormat(TextView textView, Integer number) {
		textView.setText(number.toString());
	}

	void concatenation(TextView textView, String prefix, String suffix) {
		textView.setText(prefix + suffix);
	}
}

File: Internationalization/SetTextI18n/src/main/java/net/twisterrob/lints/internationalization/SetTextI18nViolation.kt
package net.twisterrob.lints.internationalization

import android.widget.TextView

fun literal(textView: TextView) {
	textView.text = "Hello"
}

fun numberFormat(textView: TextView, number: Int) {
	textView.text = number.toString()
}

fun concatenation(textView: TextView, prefix: String, suffix: String) {
	textView.text = prefix + suffix
}

fun concatenationTemplate(textView: TextView, prefix: String, suffix: String) {
	textView.text = "$prefix$suffix"
}
